src/colorizer/CanvasOverlay.ts

```diff
@@ -467,7 +472,13 @@ export default class CanvasOverlay implements IRenderCanvas {
     this.ctx.imageSmoothingEnabled = false;
 
     if (doesInnerCanvasNeedRender || this.isExporting) {
+      // Temporarily disable the render callback to avoid re-rendering the outer
+      // canvas.
+      this.innerCanvas.setOnRenderCallback(null);
       this.innerCanvas.render(this.isExporting);
+      // Reenable after a delay because the inner canvas may be rendering
+      // asynchronously on an animation frame
+      setTimeout(() => this.innerCanvas.setOnRenderCallback(this.onInnerCanvasRender), 10);
```

Comment:
is this an arbitrary wait time? there isn't a way to make sure the render has finished?

Reply:
Edited this! There's now a function `disableCanvasSyncUntilNextRender` that handles the disable/reenable behavior, and it no longer uses an arbitrary timeout.